website/static/js/share/stats.js

```diff
@@ -79,20 +87,108 @@ function timeGraph (data) {
+    var dateHistogramAgg = {
+        sourcesByTimes: utils.termsFilter('field', '_type')
+    };
+    dateHistogramAgg.sourcesByTimes.aggregations = {
+        articlesOverTime : {
+            filter: utils.rangeFilter('providerUpdatedDateTime', threeMonthsAgo),
+            aggregations: {
+                articlesOverTime: utils.dateHistogramFilter('providerUpdatedDateTime', threeMonthsAgo)
+            }
+        }
+    };
+    return {aggregations: dateHistogramAgg};
+};
+
+/* Helper function for dealing with epoch times returned by elasticsearch */
+Stats.timeSinceEpochInMsToMMYY = function (timeSinceEpochInMs) {
+    var d = new Date(0);
+    d.setUTCSeconds(timeSinceEpochInMs / 1000);
+    return d.getMonth().toString() + '/' + d.getFullYear().toString().substring(2);
+};
+
+/* Parses elasticsearch data so that it can be fed into a c3 donut graph */
+Stats.shareDonutGraphParser = function (data) {
+    var chartData = {};
+    chartData.name = 'shareDonutGraph';
+    chartData.columns = [];
+    chartData.colors = {};
+    chartData.type = 'donut';
+
+    var providerCount = 0;
+    var hexColors = utils.generateColors(data.aggregations.sources.buckets.length);
+    var i = 0;
+    data.aggregations.sources.buckets.forEach(
+        function (bucket) {
+            chartData.columns.push([bucket.key, bucket.doc_count]);
+            providerCount = providerCount + (bucket.doc_count ? 1 : 0);
+            chartData.colors[bucket.key] = hexColors[i];
+            i = i + 1;
+        }
+    );
+    chartData.title = providerCount.toString() + ' Provider' + (providerCount !== 1 ? 's' : '');
+    $('.c3-chart-arcs-title').text(chartData.title); //dynamically update chart title
+    return chartData;
+};
+
+/* Parses elasticsearch data so that it can be fed into a c3 time graph */
+Stats.shareTimeGraphParser = function (data) {
+    var chartData = {};
+    chartData.name = 'shareTimeGraph';
+    chartData.columns = [];
+    chartData.colors = {};
+    chartData.type = 'area-spline';
+    chartData.x = 'x';
+    chartData.groups = [];
+    var grouping = [];
+    grouping.push('x');
+    var hexColors = utils.generateColors(data.aggregations.sourcesByTimes.buckets.length);
+    var i = 0;
+    var datesCol = [];
+    data.aggregations.sourcesByTimes.buckets.forEach( //TODO @bdyetton what would be nice is a helper function to do this for any agg returned by elastic
+        function (source) {
+            var total = 0;
+            chartData.colors[source.key] = hexColors[i];
+            var column = [source.key];
+            grouping.push(source.key);
+            source.articlesOverTime.articlesOverTime.buckets.forEach(function(date){
+                total = total + date.doc_count;
+                column.push(total);
+                if (i === 0) {
+                    datesCol.push(date.key);
+                }
+            });
+            chartData.columns.push(column);
+            i = i + 1;
```

Comment:
This is unnecessary; the second argument passed to the calback of `forEach` is the current index.